controller/aks-cluster-config-handler.go

```diff
@@ -411,16 +392,16 @@ func GetClusterKubeConfig(ctx context.Context, secretsCache wranglerv1.SecretCac
 }
 
 // BuildUpstreamClusterState creates AKSClusterConfigSpec from existing cluster configuration
-func BuildUpstreamClusterState(ctx context.Context, secretsCache wranglerv1.SecretCache, spec aksv1.AKSClusterConfigSpec) (*aksv1.AKSClusterConfigSpec, error) {
+func BuildUpstreamClusterState(ctx context.Context, secretsCache wranglerv1.SecretCache, spec *aksv1.AKSClusterConfigSpec) (*aksv1.AKSClusterConfigSpec, error) {
 	upstreamSpec := &aksv1.AKSClusterConfigSpec{}
 
 	upstreamSpec.Imported = true
```

Comment:
not related to this PR, but why is this always true?

Reply:
This confused me too, thank you for pointing this